test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -48,5 +60,89 @@ test.describe(
 <!-- /wp:navigation -->`
 			);
 		} );
+
+		test( 'default to my only existing menu', async ( {
+			editor,
+			navBlockUtils,
+		} ) => {
+			const createdMenu = await navBlockUtils.createNavigationMenu( {
+				title: 'Test Menu 1',
+				content:
+					'<!-- wp:navigation-link {"label":"WordPress","type":"custom","url":"http://www.wordpress.org/","kind":"custom","isTopLevelLink":true} /-->',
+			} );
+
+			await editor.insertBlock( { name: 'core/navigation' } );
+
+			// Check the markup of the block is correct.
+			await editor.publishPost();
+			const content = await editor.getEditedPostContent();
+			expect( content ).toBe(
+				`<!-- wp:navigation {"ref":${ createdMenu.id }} /-->`
+			);
+			await editor.page
+				.locator( 'role=button[name="Close panel"i]' )
+				.click();
+
+			//check the block in the canvas.
+			await expect(
+				editor.page.locator(
+					'role=gridcell[name="Custom Link link"] >> a:has-text("WordPress")'
```

Comment:
Could we use role selector for the second part here? (`a:has-text("WordPress")`)

Reply:
I don't think the link had a role I could target, but I'll double check!

Reply:
I checked and this is the wrong selector! it's actually targeting the list view on the sidebar, and that uncovered another bug, so thanks for pointing me in this direction

Reply:
Changed with the proper role selector